### PR TITLE
create terraform code for aurora db

### PR DESCRIPTION
**Pull request by hsy3418**
- Create a terraform code for creating aws aurora db
- the terraform/main.tf file is used to set up the modules, it is expecting a vpc modules, the commented statements within vpc module could be replaced by the vpc modules variables.

### CODE REVIEW

File: terraform/rds/rds.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
+resource "aws_rds_cluster" "rdsclu" {
+    cluster_identifier      = var.cluster_name
+    engine                  = "aurora-mysql"
+    engine_version          = "5.7.mysql_aurora.2.07.2"
+    db_subnet_group_name    = var.db_subnet_group_name
+    vpc_security_group_ids  = [aws_security_group.rds-apps.id]
+    availability_zones      = var.availability_zones
+    database_name           = var.db_name
+    master_username         = "admin"
+    master_password         = "admin" # the password will be repalced when security manager set up
```
Comment (caiocezart) [inline terraform/rds/rds.tf]: make those username/password variables

File: terraform/rds/rds.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    deletion_protection     = false
+    apply_immediately       = true
+    skip_final_snapshot     = true
+}
+
+resource "aws_rds_cluster_instance" "cluster_instances" {
+    count                  = 1
+    identifier             = "aurora-cluster-instance"
+    cluster_identifier     = aws_rds_cluster.rdsclu.id
+    instance_class         = "db.r4.large" # this is the smallest instance can be use for the engine
```
Comment (caiocezart) [inline terraform/rds/rds.tf:21]: you sure you can't use db.t3.small ?

File: terraform/rds/rds.tf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    engine                 = "aurora-mysql"
+    engine_version         = "5.7.mysql_aurora.2.07.2" #serverless is not supported for this version
```
Comment (caiocezart) [inline terraform/rds/rds.tf:23]: can you try and see if Wordpress image connects to it with:
engine                  = "aurora"
engine_mode             = "serverless"

**Review by caiocezart — CHANGES_REQUESTED** [review]
good work just two small notes

File: terraform/rds/rds.tf (hunk 3 above)
Comment (hsy3418) [reply terraform/rds/rds.tf:23]: I tried serverless, but got an error said serverless is not supported for this engine version, so instead I used the default mode, probably I can try use serverless with some other versions.

Comment (caiocezart) [reply terraform/rds/rds.tf:23]: that combination of `aurora` and `serverless` should do the trick

File: terraform/rds/rds.tf (hunk 2 above)
Comment (hsy3418) [reply terraform/rds/rds.tf:21]: Yes, db.t3.small is not supported for the aurora db this engine version, got an error said the combinations of db.t3.small and 5.7.mysql_aurora.2.07.2 is not supported, so I checked the console of creating the db, and "db.r4.large" seems like the smallest available option.

Comment (caiocezart) [reply terraform/rds/rds.tf:21]: if that is too expensive might be just better to switch to RDS.. I will leave that with you, I am just suggesting and discussing, not imposing anything

Comment (hsy3418) [reply terraform/rds/rds.tf:21]: Sure, I just tested it and deleted it straight away, let's see.

File: terraform/rds/rds.tf (hunk 3 above)
Comment (hsy3418) [reply terraform/rds/rds.tf:23]: I tried this, but not working, got an error `Error: error creating RDS cluster: InvalidParameterValue: The engine mode serverless you requested is currently unavailable.
        status code: 400, request id: 48d4b88e-40bc-4581-a801-38875e3203f0`

**Review by caiocezart — APPROVED** [review]
excelent work 👏  feel free to merge your PR

**Comment by caiocezart** [comment]
this fix #6

**Pull request merged**